examples/miscellaneous/miscellaneous_ex14/miscellaneous_ex14.C

```diff
@@ -0,0 +1,685 @@
+                  // temp=dphi[i][qp]*dphi[j][qp].
+                  temp = (dweight[qp]*phi[i][qp]+weight[qp]*(dphi[i][qp]-ik*dphase[qp]*phi[i][qp]))
+                    *(dphi[j][qp]+ik*dphase[qp]*phi[j][qp]);
+          
+                  // assemble the Hamiltonian: H=1/2 Nabla^2 + V
+                  H(i,j) += JxW[qp]*(0.5*temp + potval*weight[qp]*phi[i][qp]*phi[j][qp]);
+                  // assemble the mass matrix:
+                  M(i,j)+= JxW[qp]*weight[qp]*phi[i][qp]*phi[j][qp];
+                }
+            }
+        }
+      // On an unrefined mesh, constrain_element_matrix does
+      // nothing.  If this assembly function is ever repurposed to
+      // run on a refined mesh, getting the hanging node constraints
+      // right will be important.  Note that, even with
+      // asymmetric_constraint_rows = false, the constrained dof
+      // diagonals still exist in the matrix, with diagonal entries
+      // that are there to ensure non-singular matrices for linear
+      // solves but which would generate positive non-physical
+      // eigenvalues for eigensolves.
+      dof_map.constrain_element_matrix(M, dof_indices, false);
+      dof_map.constrain_element_matrix(H, dof_indices, false);
+
+      // Finally, simply add the element contribution to the
+      // overall matrices.
+      matrix_A.add_matrix (H, dof_indices);
+      matrix_B.add_matrix (M, dof_indices);
+
+    } // end of element loop
+         
+  /**
+   * All done!
+   */
+
+#else
+  // Avoid unused variable warnings when compiling without infinite
+  // elements enabled.
+  libmesh_ignore(es);
+  libmesh_ignore(system_name);
+#endif // LIBMESH_HAVE_SLEPC
+#else
+
+  // Avoid unused variable warnings when compiling without infinite
+  // elements enabled.
+  libmesh_ignore(es);
+  libmesh_ignore(system_name);
+#endif // LIBMESH_ENABLE_INFINITE_ELEMENTS
+}
+
+#ifdef LIBMESH_HAVE_SLEPC
+//define the functions needed for the @ SlepcSolverConfiguration.
+void SlepcSolverConfiguration::configure_solver()
+{
+  PetscErrorCode ierr = 0;
+
+  // if a spectral transformation was requested
+  if (_st!=INVALID_ST)
+    {   
+      // initialise the st with the default values and change the spectral transformation value.
+      ST st;
+      ierr = EPSGetST(_slepc_solver.eps(), &st);
+      libmesh_assert(ierr == 0);
+
+      // Set it to the desired type of spectral transformation.
+      // The value of the respective shift is chosen to be the target
+      // specified via \p set_position_of_spectrum().
+      switch (_st)
+        {
+        case SHIFT:
+          ierr = STSetType(st, STSHIFT);
+          break;
+        case SINVERT:
+          // this method has been renamed in 3.1
+#if SLEPC_VERSION_LESS_THAN(3,1,0)
+          ierr = STSetType(st, STSINV);
+#else
+          ierr = STSetType(st, STSINVERT);
+#endif
+          break;
+        case CAYLEY:
+#if SLEPC_VERSION_LESS_THAN(2,2,1)
```

Comment:
Could this case get turned into a libmesh_example_requires() in main?

Thanks for these, by the way; we currently have no test coverage for older SLEPc, but that doesn't mean we have no users stuck with older SLEPc.

Reply:
The oldest ersion of PETSc we still support is 2.3.3, so SLEPc support should not need to extend prior to that either...

Reply:
yes, I can do so but in fact, we use the shift-and-invert (STSINV); with this respect, the doc was wrong so I would change it and keep the version-error where it is, if it is okay for you.

Reply:
@jwpeterson has a point - we can probably get rid of the LESS_THAN(2,2,1) block entirely.

Reply:
sure.